pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -49,10 +49,9 @@ public void register(OpChain operatorChain) {
     Future<?> scheduledFuture = _executorService.submit(new TraceRunnable() {
       @Override
       public void runJob() {
-        boolean isFinished = false;
         TransferableBlock returnedErrorBlock = null;
         Throwable thrown = null;
-        try {
+        try (OpChain closeMe = operatorChain) {
```

Comment:
nit: let's add a Javadoc to this `register` method stating that the `OpChain` will be closed after execution completes?

Reply:
That is a good idea. Ideally, we should try to refactor this code to make it easier to understand.